### ISSUE TITLE
Semantic labels

### ISSUE THREAD

**Issue by qo4on** [IIGROUP/TediGAN#25]
Hi!
Where do you get such hardly visible input semantic labels like this one:
https://github.com/IIGROUP/TediGAN/blob/main/ext/experiment/images/lab/input_label.png

**Comment by qo4on** [comment]
Thanks for the clarification. All I want to do is test your `psp_celebs_seg_to_face.pt` model on the image that was not in the training CelebAMask-HQ dataset. For example, this Colab code can convert an arbitrary face image to a segmentation map. How should I modify this segmentation map so that your model can understand it?
```
!git clone https://github.com/kritiksoman/GIMP-ML.git &> /dev/null
%cd GIMP-ML
!git checkout --track origin/GIMP3-ML
%cd ..
!pip install GIMP-ML/.

import gimpml
gimpml.setup_python_weights()
%cd /content/GIMP-ML/testscases

import cv2
import matplotlib.pyplot as plt
import torch

face = cv2.imread('sampleinput/portrait.jpg')[:, :, ::-1]
out = gimpml.parseface(face)

fig = plt.figure(figsize=(10, 7))
rows, columns = 1, 2

fig.add_subplot(rows, columns, 1)
plt.imshow(face)
plt.axis('off')
plt.title("Original")

fig.add_subplot(rows, columns, 2)
plt.imshow(out)
plt.axis('off')
plt.title("Parsing")

cv2.imwrite('output/tmp-parseface.png', out[:, :, ::-1])
torch.cuda.empty_cache()
```
![image](https://user-images.githubusercontent.com/35609308/139912819-47219b79-95bf-40e5-80da-0fe17a34fad6.png)

**Comment by weihaox** [comment]
The real output of the segmentation models is ["parsing"](https://github.com/kritiksoman/GIMP-ML/blob/master/gimp-plugins/face-parsing-PyTorch/test.py#L16) not the colour one. The colour image on the right is for visualization, which can be found [here](https://github.com/kritiksoman/GIMP-ML/blob/master/gimp-plugins/face-parsing-PyTorch/test.py#L16) (I am not sure what specific segmention method you are using but the this part is similar).

**Comment by qo4on** [comment]
One more thing. Is there a way to generate several different images per each segmentation map? `n_outputs_to_generate` doesn't seem to work in `inference.py`.

**Comment by weihaox** [comment]
Sure. To generate diverse results from each segmentation map, just replace the top layers of the latent codes with those randomly-sampled ones. Please use [`--latent_mask`](https://github.com/IIGROUP/TediGAN/blob/main/ext/utils/test_options.py) to choose which layers to mix and keep in mind that replace the top layers only so that the shapes defined by given maps can be preserved.
